importer/cusource/remove_files_by_sha1.go

```diff
@@ -0,0 +1,151 @@
+package cusource
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"database/sql"
+	"fmt"
+	"github.com/Bnei-Baruch/mdb/common"
+	"github.com/Bnei-Baruch/mdb/models"
+	"github.com/Bnei-Baruch/mdb/utils"
+	log "github.com/Sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/volatiletech/sqlboiler/boil"
+	"github.com/volatiletech/sqlboiler/queries"
+	"github.com/volatiletech/sqlboiler/queries/qm"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func RemoveFilesBySHA1() {
+	log.SetLevel(log.InfoLevel)
+	mdb, err := sql.Open("postgres", viper.GetString("mdb.url"))
+	defer mdb.Close()
+	utils.Must(err)
+	utils.Must(mdb.Ping())
+	boil.SetDB(mdb)
+	boil.DebugMode = true
+	utils.Must(common.InitTypeRegistries(mdb))
+
+	run(mdb)
+}
+
+func run(mdb *sql.DB) {
+	names := getNames()
+	log.Infof("Take file names from tar: %v\n", names)
+	// actual removal
+	files, err := models.Files(mdb,
+		qm.WhereIn("name in ?", utils.ConvertArgsString(names)...),
+		qm.Load("Operations"),
+	).All()
+	utils.Must(err)
+	log.Infof("Found %d files for remove", len(files))
+	removeCount := len(files)
+	for _, f := range files {
+		oIds := make([]string, 0)
+		for _, o := range f.R.Operations {
+			oIds = append(oIds, strconv.FormatInt(o.ID, 10))
+		}
+		err = deleteFileOnTransaction(mdb, f.ID, oIds)
+		if err != nil {
+			log.Infof("File uid %s was not removed", f.UID)
+			removeCount--
+			continue
+		}
+	}
+	log.Infof("All procceses ended. Removed %d files", removeCount)
+}
+func deleteFileOnTransaction(mdb *sql.DB, fId int64, oIds []string) error {
+	log.Info("Open transaction")
+	tx, err := mdb.Begin()
+	defer func() {
+		if err == nil {
+			log.Info("Send commit of transaction ")
+			err = tx.Commit()
+			if err != nil {
+				log.Infof("Commit error %s", err)
+			}
+		} else {
+			log.Info("Send Rollback of transaction ")
+			err = tx.Rollback()
+			if err != nil {
+				log.Infof("Commit error %s", err)
+			}
+		}
+	}()
+
+	if err != nil {
```

Comment:
error handling should be immediate after error happened. This should be before the `defer`